test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,528 @@
+        .get("resource")
+        .and_then(|resource_value| resource_value.get("message"))
+        .and_then(|resource_value| resource_value.as_str())
+    {
+        uresource.message = Some(resource_value.to_owned());
+    } else {
+        error!("Error: message field is not a string in resource_value");
+        //Some("None".to_owned())
+    };
+
+    if let Some(resource_value) = value
+        .get("resource")
+        .and_then(|resource_value| resource_value.get("id"))
+        .and_then(|resource_value| resource_value.as_str())
+    {
+        if let Ok(parsed_id) = resource_value.parse::<u32>() {
+            uresource.id = Some(parsed_id);
+        } else {
+            error!("Error: id field parsing to u32");
+            // Some(0)
+        }
+    } else {
+        error!("Error: id field is not string");
+        //Some(0)
+    };
+    uresource
+}
+
+fn parse_string_field(value: &Value, field: &str) -> Result<String, serde_json::Error> {
+    if let Some(entity_value) = value
+        .get("entity")
+        .and_then(|entity_value| entity_value.get(field))
+        .and_then(|field_value| field_value.as_str())
+    {
+        Ok(entity_value.to_owned())
+    } else {
+        error!("Error: {field} field is not a string");
+        Err(serde::de::Error::custom(format!(
+            "Error: {field} field is not a string"
+        )))
+    }
+}
+
+fn parse_u32_field(value: &Value, field: &str) -> Result<Option<u32>, serde_json::Error> {
+    if let Some(entity_value) = value
+        .get("entity")
+        .and_then(|entity_value| entity_value.get(field))
+        .and_then(|field_value| field_value.as_str())
+    {
+        if let Ok(parsed_value) = entity_value.parse::<u32>() {
+            Ok(Some(parsed_value))
+        } else {
+            error!("Error: {field} is not a number");
+            Err(serde::de::Error::custom(format!(
+                "Error: {field} is not a number"
+            )))
+        }
+    } else {
+        error!("Error: {} field is not a string", field);
+        Err(serde::de::Error::custom(format!(
+            "Error: {field} field is not a string"
+        )))
+    }
+}
+
+fn parse_uentity(value: &Value) -> UEntity {
+    let name = match parse_string_field(value, "name") {
+        Ok(value) => value,
+        Err(_) => "None".to_owned(),
+    };
+
+    let id = match parse_u32_field(value, "id") {
+        Ok(value) => value,
+        Err(_) => None,
+    };
+
+    let version_major = match parse_u32_field(value, "version_major") {
+        Ok(value) => value,
+        Err(_) => None,
+    };
+
+    let version_minor = match parse_u32_field(value, "version_minor") {
+        Ok(value) => value,
+        Err(_) => None,
+    };
+
+    UEntity {
+        name,
+        id,
+        version_major,
+        version_minor,
+        special_fields: SpecialFields::default(),
+    }
+}
+
+fn parse_uauthority(value: &Value) -> Result<UAuthority, serde_json::Error> {
```

Comment:
This signature feels a little off to me.

I'd expect to see this instead:
```rust
fn parse_uauthority(value: &Value) -> Option<UAuthority>
```
given it's valid for a UUri not have no UAuthority.

Where `parse_uauthority()` is called, the error is not made use of in any case, so I think you could replace where an `Err` is being returned to just outputting an `error!()` and return `None`.